unit_test/main_tests/main_platform_test.py

```diff
@@ -10,25 +10,26 @@
 from ..conftest import MOCK_PACKAGE_DIR
 
 
-def test_unknown_platform_non_ci(monkeypatch, capsys):
-    monkeypatch.delenv("CI", raising=False)
-    monkeypatch.delenv("BITRISE_BUILD_NUMBER", raising=False)
-    monkeypatch.delenv("AZURE_HTTP_USER_AGENT", raising=False)
-    monkeypatch.delenv("TRAVIS", raising=False)
-    monkeypatch.delenv("APPVEYOR", raising=False)
-    monkeypatch.delenv("GITHUB_ACTIONS", raising=False)
-    monkeypatch.delenv("GITLAB_CI", raising=False)
-    monkeypatch.delenv("CIRCLECI", raising=False)
-    monkeypatch.delenv("CIRRUS_CI", raising=False)
-    monkeypatch.delenv("CIBW_PLATFORM", raising=False)
+@pytest.mark.parametrize("option_value", [None, "auto"])
+def test_platform_unset_or_auto(monkeypatch, intercepted_build_args, option_value):
+    if option_value is None:
+        monkeypatch.delenv("CIBW_PLATFORM", raising=False)
+    else:
+        monkeypatch.setenv("CIBW_PLATFORM", option_value)
 
-    with pytest.raises(SystemExit) as exit:
-        main()
-    assert exit.value.code == 2
-    _, err = capsys.readouterr()
+    main()
 
-    assert "cibuildwheel: Unable to detect platform." in err
-    assert "cibuildwheel should run on your CI server" in err
+    options = intercepted_build_args.args[0]
+
+    # check that the platform was auto detected to build for the current system
+    if sys.platform.startswith("linux"):
```

Comment:
nit: this feels like too much logic in a test. Any chance to simplify it?

Reply:
I know what you mean, it's essentially just repeating the logic in the main program. But I would like to be sure that both `auto` and unspecified are doing the right thing. A little extra verbosity in the test seems like a small price to pay IMO.

Reply:
In that case, why not try solving this on the level of parametrization? It'd be even clearer if it shows up in the test report.

Reply:
Even not parametrization but simeple dict could solve this:
```python
platform_dkt = {"linux": "linux", "darwin": "macos", "win32": "windows"}

assert sys.platform in platform_dkt, f"Unknown platform: {sys.platform}"
assert platform_dkt[sys.platform] == options.platform
```

Reply:
While I appreciate the focus on craft and code quality, there becomes a point when it's not worth our time to debate such minor points. Especially in tests, where the control flow complexity is so low, because each test is an isolated execution. All that matters in a test is the actions that are taken and the asserts that are reached - if these are clear to the reader, it's fine. If they grow to become unmaintainable, it's easy to rewrite them.